third-party/linshare/src/test/java/org/apache/james/linshare/LinshareDockerCompose.java

```diff
@@ -0,0 +1,34 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.linshare;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+interface LinshareDockerCompose {
```

Comment:
Why is this interface needed? What does it add?

Reply:
What I want is provide an Linshare Interface to the client, where it can be a normal linshare or singleton linshare. With singleton linshare, I can override start() & stop() to do nothing while with normal linshare, it acts start() & stop() normally. 

Clients can have a choice to choose which impl it interests while still be able to use a same suit of apis.